utils/modular_model_converter.py

```diff
@@ -748,11 +751,52 @@ def leave_Module(self, original_node: cst.Module, node):
         nodes_to_remove = [
             self.top_level_functions_or_classes[name] for name in unused if name in self.top_level_functions_or_classes
         ]
-        new_body = [node_ for node_ in original_node.body if node_ not in nodes_to_remove]
+        new_body = [node_ for node_ in node.body if node_ not in nodes_to_remove]
         # Return a new module with the updated body
         return node.with_changes(body=new_body)
 
 
+    def leave_If(self,original_node: cst.If,updated_node: cst.If):
+        for stmt in original_node.body.body:
+            if m.matches(stmt, m.SimpleStatementLine(body=[m.ImportFrom() | m.Import()])):
+                if len(updated_node.body.body) == 0:
+                    return cst.RemoveFromParent()
+        return updated_node
+
+    @m.leave(m.Import() | m.ImportFrom())
+    def leave_import_alike(self, original_node, updated_node):
+        names_to_keep = []
+        for name in updated_node.names:
+            name_value = name.evaluated_name
+            if name_value not in self.unused_imports:
+                names_to_keep.append(name.with_changes(comma=cst.MaybeSentinel.DEFAULT))
+        if len(names_to_keep) == 0:
+            return cst.RemoveFromParent()
+        else:
+            return updated_node.with_changes(names=names_to_keep)
+
+
+
+def get_unused_imports(source):
+    wrapper = cst.metadata.MetadataWrapper(source)
+    scopes = set(wrapper.resolve(cst.metadata.ScopeProvider).values())
+    unused_imports: Dict[Union[cst.Import, cst.ImportFrom], Set[str]] = defaultdict(set)
```

Comment:
Wrong type here, should be `Dict[str, Set[Union[cst.Import, cst.ImportFrom]]]` if I understand correctly